src/pip/_internal/download.py

```diff
@@ -598,6 +598,9 @@ def __init__(self, *args, **kwargs):
     def add_insecure_host(self, host):
         # type: (str) -> None
         self.mount('https://{}/'.format(host), self._insecure_adapter)
+        if ":" not in host:
```

Comment:
Shouldn't you be using some parsing function (e.g. based on `urllib_parse.urlparse()`)? It doesn't seem like this would work e.g. with IPv6 addresses.

Reply:
Yes, I tried the function but it didn't work for the scheme missing. Is there a more elegant way to do this than supplying the scheme to form a full URL?

Reply:
Well, a scheme is being added above anyways, so it wouldn’t require doing more than what’s already being done.

Reply:
Hmmm, things get complicated when counting with IPv6 addresses, need a utility function to reliably construct a URL with IPv6 host(and potentially port).